websites/G/GG App/presence.ts

```diff
@@ -0,0 +1,134 @@
+const presence = new Presence({
+    clientId: "928868301105426463"
+  }),
+  browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+      largeImageKey: "lg",
+      startTimestamp: browsingTimestamp
+    },
+    path = document.location.pathname,
+    [timestamps, cover, buttons] = await Promise.all([
+      presence.getSetting<boolean>("timestamps"),
+      presence.getSetting<boolean>("cover"),
+      presence.getSetting<boolean>("buttons")
+    ]);
+  // Apparently "/games" is a user page so .pathname can't be used to detect a game but instead the header
+  if (
+    document.querySelector<HTMLHeadingElement>(
+      "#root > div > div.css-1dbjc4n > div > div.w-full.xl\\:max-w-max.mx-auto.-mt-64.sm\\:-mt-48.px-4 > div.flex.flex-col.md\\:flex-row.pt-8 > div.w-full.md\\:w-3\\/4.md\\:pl-4.lg\\:pl-8 > div:nth-child(1) > h1"
+    ) &&
+    path.startsWith("/games")
+  ) {
+    presenceData.details = "Viewing Game:";
+    presenceData.state = document.querySelector<HTMLHeadingElement>(
+      "#root > div > div.css-1dbjc4n > div > div.w-full.xl\\:max-w-max.mx-auto.-mt-64.sm\\:-mt-48.px-4 > div.flex.flex-col.md\\:flex-row.pt-8 > div.w-full.md\\:w-3\\/4.md\\:pl-4.lg\\:pl-8 > div:nth-child(1) > h1"
+    ).textContent;
+    if (cover) {
+      presenceData.largeImageKey = document.querySelector<HTMLImageElement>(
+        "#root > div > div.css-1dbjc4n > div > div.w-full.xl\\:max-w-max.mx-auto.-mt-64.sm\\:-mt-48.px-4 > div.flex.flex-col.md\\:flex-row.pt-8 > div.css-1dbjc4n > img"
+      ).src;
+      presenceData.smallImageKey = "lg";
+    }
+    presenceData.buttons = [{ label: "View Game", url: document.URL }];
+  } else if (
+    document.querySelector<HTMLHeadingElement>(
+      "#root > div > div.css-1dbjc4n > div > div:nth-child(3) > div.w-full > div > div.flex.flex-col.md\\:mx-4.xl\\:mx-8.self-center.md\\:self-start > div:nth-child(2) > h2"
+    ) ||
+    document.querySelector<HTMLAnchorElement>(
+      "#root > div > div.css-1dbjc4n > div > div.w-full.xl\\:max-w-max.mx-auto.pt-8 > div > div.flex.flex-column.p-3 > div > div.flex.flex-column.md\\:flex-row.items-start.md\\:items-center > p.text-white.md\\:mt-2 > a"
+    )
+  ) {
+    const username = document.querySelector<HTMLAnchorElement>(
+        "#root > div > div.css-1dbjc4n > div > div.w-full.xl\\:max-w-max.mx-auto.pt-8 > div > div.flex.flex-column.p-3 > div > div.flex.flex-column.md\\:flex-row.items-start.md\\:items-center > p.text-white.md\\:mt-2 > a"
```

Comment:
Where is this not null?

Reply:
It was meant to check if the profile picture was located in different locations for different profiles because it looked like so to me.
So far, no null. So I can delete that if you want, that is just a safeguard.

Reply:
To differentiate this
![image](https://user-images.githubusercontent.com/77577746/148675772-e8684209-c7f5-4097-9786-2ed9f638f9a2.png)
And this
![image](https://user-images.githubusercontent.com/77577746/148675783-fa9b850f-845b-45c5-929d-2797d2d15df5.png)